### PR TITLE
Remove economic protocol handling

### PR DESCRIPTION
**Pull request by ureeves**
These commits remove the handling of the economic protocol introduced previously, as decided by team consensus. It also takes the liberty of upgrading the `piecrust` stack to the newest versions, which cost more gas due to the way storage is handled.

### CODE REVIEW

File: rusk-abi/Cargo.toml [hunk 1]
```diff
@@ -17,8 +17,8 @@ dusk-bytes = "0.1"
 bytecheck = { version = "0.6", default-features = false }
 dusk-plonk = { version = "0.19", default-features = false, features = ["rkyv-impl", "alloc"] }
 
-piecrust-uplink = { version = "0.12" }
-piecrust = { version = "0.19", optional = true }
```
Comment (xevisalle) [inline rusk-abi/Cargo.toml:21]: should the rusk-abi CHANGELOG reflect these changes?

Comment (ureeves) [reply rusk-abi/Cargo.toml:21]: Yes, you're right. Updated

**Review by HDauven — DISMISSED** [review]

**Review by herr-seppia — APPROVED** [review]
LGTM

**Review by xevisalle — APPROVED** [review]

**Review by xevisalle — APPROVED** [review]

**Review by moCello — APPROVED** [review]
LGTM

**Pull request merged**